doschema/cli.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoMapping.
+# Copyright (C) 2015, 2016 CERN.
+#
+# DoMapping is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoMapping is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoMapping; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""CLI commands."""
+
+import json
+import os
+from io import open
+
+import chardet
+import click
+
+import doschema.validation
+
+
+@click.group()
+def cli():
+    """CLI group."""
+    pass  # pragma: no cover
+
+
+@cli.command()
+@click.argument('schemas',
+                type=click.Path(
+                    exists=True, file_okay=True, dir_okay=False
+                ),
+                nargs=-1)
+@click.option('--ignore_index/--with_index',
+              default=True, help='Disable ignoring array indexes')
+def validate(schemas, ignore_index):
+    """Main function for cli."""
+    schema_validator = doschema.validation.JSONSchemaValidator(
+        ignore_index)
+    for schema in schemas:
+        path = os.path.join(os.getcwd(), 'tests', 'data_files', schema)
+        with open(path, 'rb') as infile:
+            byte_file = infile.read()
+            encoding = chardet.detect(byte_file)['encoding']
+            string_file = byte_file.decode(encoding)
+            schema = json.loads(
+                string_file,
+                encoding=encoding
```

Comment:
I don't think that specifying the encoding here is useful. the string has already been decoded.

However I just tested how `chardet` and `json.loads` behave with an utf-16 file. There is an issue with the BOM as chardet returns 'UTF-16BE' or 'UTF16-LE' and `bytes.decode` does not remove the BOM. It might be a good idea to remove the BOM manually and add a test to check UTF-16 files.